<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Subsampling single-cell data sets with sketchR • sketchR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Subsampling single-cell data sets with sketchR">
<meta property="og:description" content="sketchR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sketchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sketchR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Subsampling single-cell data sets with
sketchR</h1>
            
            <h4 data-toc-skip class="date">2023-06-16</h4>
      
      
      <div class="hidden name"><code>sketchR.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette showcases the main functionalities of the
<code>sketchR</code> package, and illustrates how it can be used to
generate a subsample of a dataset using the geometric
sketching/subsampling algorithms and implementations proposed by <span class="citation">Hie et al. (2019)</span> and <span class="citation">Song et al. (2022)</span>, as well as create a set of
diagnostic plots.</p>
</div>
<div class="section level2">
<h2 id="preparation">Preparation<a class="anchor" aria-label="anchor" href="#preparation"></a>
</h2>
<p>We start by loading the required packages and preparing an example
data set.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">sketchR</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scuttle</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR" class="external-link">SingleR</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/celldex" class="external-link">celldex</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>We will use the PBMC3k data set from the <em><a href="https://bioconductor.org/packages/3.17/TENxPBMCData" class="external-link">TENxPBMCData</a></em>
Bioconductor package for illustration. The chunk below prepares the data
set by calculating log-transformed normalized counts, finding highly
variable genes, performing dimensionality reduction and predicting cell
type labels using the <em><a href="https://bioconductor.org/packages/3.17/SingleR" class="external-link">SingleR</a></em>
package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load data</span></span>
<span><span class="va">pbmc3k</span> <span class="op">&lt;-</span> <span class="fu">TENxPBMCData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html" class="external-link">TENxPBMCData</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"pbmc3k"</span><span class="op">)</span></span>
<span><span class="co">#&gt; see ?TENxPBMCData and browseVignettes('TENxPBMCData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span></span>
<span><span class="co">## Set row and column names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Cell"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html" class="external-link">uniquifyFeatureNames</a></span><span class="op">(</span></span>
<span>    ID <span class="op">=</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL_ID</span>,</span>
<span>    names <span class="op">=</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol_TENx</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Normalize and log-transform counts</span></span>
<span><span class="va">pbmc3k</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find highly variable genes</span></span>
<span><span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html" class="external-link">modelGeneVar</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span></span>
<span><span class="va">top.hvgs</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html" class="external-link">getTopHVGs</a></span><span class="op">(</span><span class="va">dec</span>, n <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Perform dimensionality reduction</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">pbmc3k</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">pbmc3k</span>, subset_row <span class="op">=</span> <span class="va">top.hvgs</span><span class="op">)</span></span>
<span><span class="va">pbmc3k</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html" class="external-link">runTSNE</a></span><span class="op">(</span><span class="va">pbmc3k</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Predict cell type labels</span></span>
<span><span class="va">ref_monaco</span> <span class="op">&lt;-</span> <span class="fu">celldex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/celldex/man/MonacoImmuneData.html" class="external-link">MonacoImmuneData</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">pred_monaco_main</span> <span class="op">&lt;-</span> <span class="fu">SingleR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html" class="external-link">SingleR</a></span><span class="op">(</span>test <span class="op">=</span> <span class="va">pbmc3k</span>, ref <span class="op">=</span> <span class="va">ref_monaco</span>, </span>
<span>                                     labels <span class="op">=</span> <span class="va">ref_monaco</span><span class="op">$</span><span class="va">label.main</span><span class="op">)</span></span>
<span><span class="va">pbmc3k</span><span class="op">$</span><span class="va">labels_main</span> <span class="op">&lt;-</span> <span class="va">pred_monaco_main</span><span class="op">$</span><span class="va">labels</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 32738  2700</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="subsampling">Subsampling<a class="anchor" aria-label="anchor" href="#subsampling"></a>
</h2>
<p>The <code><a href="../reference/geosketch.html">geosketch()</a></code> function performs geometric sketching by
calling the <code>geosketch</code> <a href="https://github.com/brianhie/geosketch" class="external-link">python package</a>. The
output is a vector of indices that can be used to subset the full
dataset. The provided seed will be propagated to the python code to
achieve reproducibility.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx800gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geosketch.html">geosketch</a></span><span class="op">(</span><span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">pbmc3k</span>, <span class="st">"PCA"</span><span class="op">)</span>, </span>
<span>                      N <span class="op">=</span> <span class="fl">800</span>, seed <span class="op">=</span> <span class="fl">123</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">idx800gs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1 2 5 6 7 8</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">idx800gs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 800</span></span></code></pre></div>
<p>Similarly, the <code><a href="../reference/scsampler.html">scsampler()</a></code> function calls the
<code>scSampler</code> <a href="https://github.com/SONGDONGYUAN1994/scsampler" class="external-link">python package</a>
to perform subsampling.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx800scs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scsampler.html">scsampler</a></span><span class="op">(</span><span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">pbmc3k</span>, <span class="st">"PCA"</span><span class="op">)</span>, </span>
<span>                       N <span class="op">=</span> <span class="fl">800</span>, seed <span class="op">=</span> <span class="fl">123</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">idx800scs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1079  644 1494 1278    5  391</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">idx800scs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 800</span></span></code></pre></div>
<p>To illustrate the result of the subsampling, we plot the tSNE
representation of the original data as well as the two subsets (using
the tSNE coordinates derived from the full dataset).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html" class="external-link">plotTSNE</a></span><span class="op">(</span><span class="va">pbmc3k</span>, colour_by <span class="op">=</span> <span class="st">"labels_main"</span><span class="op">)</span>,</span>
<span>    <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html" class="external-link">plotTSNE</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">[</span>, <span class="va">idx800gs</span><span class="op">]</span>, colour_by <span class="op">=</span> <span class="st">"labels_main"</span><span class="op">)</span>,</span>
<span>    <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html" class="external-link">plotTSNE</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">[</span>, <span class="va">idx800scs</span><span class="op">]</span>, colour_by <span class="op">=</span> <span class="st">"labels_main"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="sketchR_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
<p>We can also illustrate the relative abundance of each cell type in
the full data and in the subsets, respectively.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compareCompositionPlot.html">compareCompositionPlot</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">pbmc3k</span><span class="op">)</span>, </span>
<span>                       idx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>geosketch <span class="op">=</span> <span class="va">idx800gs</span>,</span>
<span>                                  scSampler <span class="op">=</span> <span class="va">idx800scs</span><span class="op">)</span>, </span>
<span>                       column <span class="op">=</span> <span class="st">"labels_main"</span><span class="op">)</span></span></code></pre></div>
<p><img src="sketchR_files/figure-html/unnamed-chunk-6-1.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="diagnostic-plots">Diagnostic plots<a class="anchor" aria-label="anchor" href="#diagnostic-plots"></a>
</h2>
<p><code>sketchR</code> provides a convenient function to plot the
Hausdorff distance between the full dataset and the subsample, for a
range of sketch sizes.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hausdorffDistPlot.html">hausdorffDistPlot</a></span><span class="op">(</span>mat <span class="op">=</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">pbmc3k</span>, <span class="st">"PCA"</span><span class="op">)</span>, </span>
<span>                  Nvec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">400</span>, <span class="fl">800</span>, <span class="fl">2000</span><span class="op">)</span>, seed <span class="op">=</span> <span class="fl">123</span>, </span>
<span>                  Nrep <span class="op">=</span> <span class="fl">3</span>, methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"geosketch"</span>, <span class="st">"scsampler"</span>, <span class="st">"uniform"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="sketchR_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.6.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats    graphics utils    stats4   methods  base    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] cowplot_1.1.1               celldex_1.10.1             </span></span>
<span><span class="co">#&gt;  [3] SingleR_2.2.0               scater_1.28.0              </span></span>
<span><span class="co">#&gt;  [5] ggplot2_3.4.2               scran_1.28.1               </span></span>
<span><span class="co">#&gt;  [7] scuttle_1.10.1              TENxPBMCData_1.18.0        </span></span>
<span><span class="co">#&gt;  [9] HDF5Array_1.28.1            rhdf5_2.44.0               </span></span>
<span><span class="co">#&gt; [11] DelayedArray_0.26.3         S4Arrays_1.0.4             </span></span>
<span><span class="co">#&gt; [13] Matrix_1.5-4.1              SingleCellExperiment_1.22.0</span></span>
<span><span class="co">#&gt; [15] SummarizedExperiment_1.30.2 Biobase_2.60.0             </span></span>
<span><span class="co">#&gt; [17] GenomicRanges_1.52.0        GenomeInfoDb_1.36.0        </span></span>
<span><span class="co">#&gt; [19] IRanges_2.34.0              S4Vectors_0.38.1           </span></span>
<span><span class="co">#&gt; [21] BiocGenerics_0.46.0         MatrixGenerics_1.12.2      </span></span>
<span><span class="co">#&gt; [23] matrixStats_1.0.0           sketchR_0.1.6              </span></span>
<span><span class="co">#&gt; [25] BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.5                magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [3] ggbeeswarm_0.7.2              farver_2.1.1                 </span></span>
<span><span class="co">#&gt;   [5] rmarkdown_2.22                fs_1.6.2                     </span></span>
<span><span class="co">#&gt;   [7] zlibbioc_1.46.0               ragg_1.2.5                   </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.3                   memoise_2.0.1                </span></span>
<span><span class="co">#&gt;  [11] DelayedMatrixStats_1.22.1     RCurl_1.98-1.12              </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.5               AnnotationHub_3.8.0          </span></span>
<span><span class="co">#&gt;  [15] curl_5.0.1                    BiocNeighbors_1.18.0         </span></span>
<span><span class="co">#&gt;  [17] Rhdf5lib_1.22.0               sass_0.4.6                   </span></span>
<span><span class="co">#&gt;  [19] bslib_0.5.0                   basilisk_1.12.0              </span></span>
<span><span class="co">#&gt;  [21] desc_1.4.2                    cachem_1.0.8                 </span></span>
<span><span class="co">#&gt;  [23] igraph_1.5.0                  mime_0.12                    </span></span>
<span><span class="co">#&gt;  [25] lifecycle_1.0.3               pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [27] rsvd_1.0.5                    R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [29] fastmap_1.1.1                 GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt;  [31] shiny_1.7.4                   digest_0.6.31                </span></span>
<span><span class="co">#&gt;  [33] colorspace_2.1-0              AnnotationDbi_1.62.1         </span></span>
<span><span class="co">#&gt;  [35] rprojroot_2.0.3               dqrng_0.3.0                  </span></span>
<span><span class="co">#&gt;  [37] irlba_2.3.5.1                 ExperimentHub_2.8.0          </span></span>
<span><span class="co">#&gt;  [39] textshaping_0.3.6             RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;  [41] beachmat_2.16.0               labeling_0.4.2               </span></span>
<span><span class="co">#&gt;  [43] filelock_1.0.2                fansi_1.0.4                  </span></span>
<span><span class="co">#&gt;  [45] httr_1.4.6                    compiler_4.3.0               </span></span>
<span><span class="co">#&gt;  [47] here_1.0.1                    bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [49] withr_2.5.0                   BiocParallel_1.34.2          </span></span>
<span><span class="co">#&gt;  [51] viridis_0.6.3                 DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [53] highr_0.10                    rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [55] bluster_1.10.0                tools_4.3.0                  </span></span>
<span><span class="co">#&gt;  [57] vipor_0.4.5                   beeswarm_0.4.0               </span></span>
<span><span class="co">#&gt;  [59] interactiveDisplayBase_1.38.0 httpuv_1.6.11                </span></span>
<span><span class="co">#&gt;  [61] glue_1.6.2                    rhdf5filters_1.12.1          </span></span>
<span><span class="co">#&gt;  [63] promises_1.2.0.1              grid_4.3.0                   </span></span>
<span><span class="co">#&gt;  [65] Rtsne_0.16                    cluster_2.1.4                </span></span>
<span><span class="co">#&gt;  [67] generics_0.1.3                gtable_0.3.3                 </span></span>
<span><span class="co">#&gt;  [69] BiocSingular_1.16.0           ScaledMatrix_1.8.1           </span></span>
<span><span class="co">#&gt;  [71] metapod_1.8.0                 utf8_1.2.3                   </span></span>
<span><span class="co">#&gt;  [73] XVector_0.40.0                ggrepel_0.9.3                </span></span>
<span><span class="co">#&gt;  [75] BiocVersion_3.17.1            pillar_1.9.0                 </span></span>
<span><span class="co">#&gt;  [77] stringr_1.5.0                 limma_3.56.2                 </span></span>
<span><span class="co">#&gt;  [79] later_1.3.1                   dplyr_1.1.2                  </span></span>
<span><span class="co">#&gt;  [81] BiocFileCache_2.8.0           lattice_0.21-8               </span></span>
<span><span class="co">#&gt;  [83] bit_4.0.5                     tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [85] locfit_1.5-9.8                Biostrings_2.68.1            </span></span>
<span><span class="co">#&gt;  [87] knitr_1.43                    gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [89] edgeR_3.42.4                  xfun_0.39                    </span></span>
<span><span class="co">#&gt;  [91] datasets_4.3.0                statmod_1.5.0                </span></span>
<span><span class="co">#&gt;  [93] stringi_1.7.12                yaml_2.3.7                   </span></span>
<span><span class="co">#&gt;  [95] evaluate_0.21                 codetools_0.2-19             </span></span>
<span><span class="co">#&gt;  [97] tibble_3.2.1                  BiocManager_1.30.21          </span></span>
<span><span class="co">#&gt;  [99] cli_3.6.1                     xtable_1.8-4                 </span></span>
<span><span class="co">#&gt; [101] reticulate_1.30               systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [103] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [105] Rcpp_1.0.10                   dir.expiry_1.8.0             </span></span>
<span><span class="co">#&gt; [107] dbplyr_2.3.2                  png_0.1-8                    </span></span>
<span><span class="co">#&gt; [109] parallel_4.3.0                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [111] pkgdown_2.0.7.9000            blob_1.2.4                   </span></span>
<span><span class="co">#&gt; [113] basilisk.utils_1.12.1         sparseMatrixStats_1.12.0     </span></span>
<span><span class="co">#&gt; [115] bitops_1.0-7                  viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [117] scales_1.2.1                  grDevices_4.3.0              </span></span>
<span><span class="co">#&gt; [119] purrr_1.0.1                   crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [121] rlang_1.1.1                   KEGGREST_1.40.0</span></span></code></pre></div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Hie2019-geosketch" class="csl-entry">
Hie, Brian, Hyunghoon Cho, Benjamin DeMeo, Bryan Bryson, and Bonnie
Berger. 2019. <span>“Geometric Sketching Compactly Summarizes the
<span>Single-Cell</span> Transcriptomic Landscape.”</span> <em>Cell
Syst</em> 8 (6): 483–493.e7.
</div>
<div id="ref-Song2022-scsampler" class="csl-entry">
Song, Dongyuan, Nan Miles Xi, Jingyi Jessica Li, and Lin Wang. 2022.
<span>“scSampler: Fast Diversity-Preserving Subsampling of Large-Scale
Single-Cell Transcriptomic Data.”</span> <em>bioRxiv</em>,
2022.01.15.476407.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte Soneson, Michael Stadler.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
